app/src/main/kotlin/com/hedvig/app/feature/marketing/data/GetInitialMarketPickerValuesUseCase.kt

```diff
@@ -9,19 +8,20 @@ import com.hedvig.android.hanalytics.featureflags.flags.Feature
 import com.hedvig.android.market.Language
 import com.hedvig.android.market.Market
 import com.hedvig.android.market.MarketManager
+import com.hedvig.app.LanguageService
 import com.hedvig.app.util.apollo.safeQuery
 import com.hedvig.app.util.apollo.toEither
 
 class GetInitialMarketPickerValuesUseCase(
   private val marketManager: MarketManager,
-  private val context: Context,
   private val apolloClient: ApolloClient,
   private val featureManager: FeatureManager,
+  private val languageService: LanguageService,
 ) {
   suspend operator fun invoke(): Pair<Market, Language> {
     val currentMarket = marketManager.market
     if (currentMarket != null) {
-      val currentLanguage = Language.fromSettings(context, currentMarket)
+      val currentLanguage = languageService.getLanguage() ?: currentMarket.defaultLanguage()
```

Comment:
Perfect, now I see why we need this to be nullable.
Can we then maybe instead do the opposite, have LanguageService be the one which knows how to handle that case by having a dependency on MarketManager so that each time we're using them we don't need to think of such logic?
Have it go smth like
1. pick selected lang
2. if none is picked, ask market for current market and get default lang
3. if there is no market fallback to en_SE
and have all this logic happen in LanguageService to make using it dead simple and always return a non-null result.